tests/operators/test_offres_exp_approx.py

```diff
@@ -0,0 +1,169 @@
+"""Test off-resonance spatial coefficient and temporal interpolator estimation."""
+
+import math
+
+import numpy as np
+import numpy.testing as npt
+
+import pytest
+
+import mrinufft
+from mrinufft.operators.base import CUPY_AVAILABLE, AUTOGRAD_AVAILABLE
+from mrinufft._utils import get_array_module
+
+from helpers import to_interface, from_interface
+
+
+def calculate_true_offresonance_term(fieldmap, t):
+    """Calculate non-approximate off-resonance modulation term."""
+    xp = get_array_module(fieldmap)
+    t = xp.asarray(t)
+    if xp.__name__ == "torch":
+        t = t.to(device=fieldmap.device)
+    arg = t * fieldmap[..., None]
+    arg = arg[None, ...].swapaxes(0, -1)[..., 0]
+    return xp.exp(-arg)
+
+
+def calculate_approx_offresonance_term(B, C):
+    """Calculate approximate off-resonance modulation term."""
+    field_term = 0.0
+    for n in range(B.shape[0]):
+        tmp = B[n] * C[n][..., None]
+        tmp = tmp[None, ...].swapaxes(0, -1)[..., 0]
+        field_term += tmp
+    return field_term
+
+
+def _make_b0map(ndim, npix=64):
```

Comment:
you could probably refactor the three function below into a Case (like the CaseTrajectory) in `tests/case_trajectories.py`